crates/networking/p2p/peer_handler.rs

```diff
@@ -441,48 +435,54 @@ impl PeerHandler {
                 skip: 0,
                 reverse: matches!(order, BlockRequestOrder::NewToOld),
             });
-            let (peer_id, mut peer_channel) = self
-                .get_peer_channel_with_retry(&SUPPORTED_ETH_CAPABILITIES)
-                .await?;
-            let mut receiver = peer_channel.receiver.lock().await;
-            if let Err(err) = peer_channel
-                .connection
-                .cast(CastMessage::BackendMessage(request))
-                .await
-            {
-                debug!("Failed to send message to peer: {err:?}");
-                continue;
-            }
-            if let Some(block_headers) = tokio::time::timeout(PEER_REPLY_TIMEOUT, async move {
-                loop {
-                    match receiver.recv().await {
-                        Some(RLPxMessage::BlockHeaders(BlockHeaders { id, block_headers }))
-                            if id == request_id =>
-                        {
-                            return Some(block_headers);
+            match self.get_random_peer(&SUPPORTED_ETH_CAPABILITIES).await? {
```

Comment:
`get_random_peer` doesn't check if the peer is used. What would happen if the peer is already serving one of our requests?

Reply:
Yeah, that seems to be a pre-existent issue. It's [even documented](https://github.com/lambdaclass/ethrex/blob/main/crates/networking/p2p/peer_handler.rs#L160)
Eventually our peers should be able to handle multiple inflight requests.

Reply:
Added an issue to review this: https://github.com/lambdaclass/ethrex/issues/4756